library/src/scripts/components/siteNav/SiteNavNode.tsx

```diff
@@ -0,0 +1,240 @@
+            this.updateCurrentState();
+        }
+    }
+
+    public componentDidMount() {
+        this.openRecursive();
+    }
+
+    public render() {
+        const hasChildren = this.props.children && this.props.children.length > 0;
+        const topLevel = this.props.counter === 1;
+        const childrenContents =
+            hasChildren &&
+            this.props.children.map((child, i) => {
+                return (
+                    <SiteNavNode
+                        {...child}
+                        key={"siteNavNode-" + this.props.counter + "-" + i}
+                        counter={this.props.counter! + 1}
+                        openParent={this.openSelfAndOpenParent}
+                        location={this.props.location}
+                        depth={this.props.depth + 1}
+                    />
+                );
+            });
+        const space = `&nbsp;`;
+        return (
+            <li
+                className={classNames("siteNavNode", this.props.className, { isCurrent: this.state.current })}
+                role="treeitem"
+                aria-expanded={this.state.open}
+                style={{ marginLeft: `${(this.props.depth - 1) * 18}px` }}
+            >
+                {hasChildren && (
+                    <Button
+                        tabIndex={-1}
+                        ariaHidden={true}
+                        title={t("Toggle Category")}
+                        ariaLabel={t("Toggle Category")}
+                        onClick={this.handleClick as any}
+                        baseClass={ButtonBaseClass.CUSTOM}
+                        className="siteNavNode-toggle"
+                    >
+                        {this.state.open ? downTriangle(t("Expand")) : rightTriangle(t("Collapse"))}
+                    </Button>
+                )}
+                {!hasChildren && (
+                    <span
+                        className="siteNavNode-spacer"
+                        aria-hidden={true}
+                        dangerouslySetInnerHTML={{ __html: space }}
+                    />
+                )}
+                <div className={classNames("siteNavNode-contents")}>
+                    <Link
+                        onKeyDownCapture={this.handleKeyDown}
+                        className={classNames("siteNavNode-link")}
+                        tabIndex={0}
+                        to={this.props.url}
+                    >
+                        <span className="siteNavNode-label">{this.props.name}</span>
+                    </Link>
+                    {hasChildren && (
+                        <ul className={classNames("siteNavNode-children", { isHidden: !this.state.open })} role="group">
+                            {childrenContents}
+                        </ul>
+                    )}
+                </div>
+            </li>
+        );
+    }
+
+    private next = (tabHandler, currentLink) => {
```

Comment:
`tabHandler` should be typed as `tabHandler: TabHandler` here. `currentLink` seems to be an `Element`.